pkg/kubelet/kubelet_node_status.go

```diff
@@ -413,6 +411,7 @@ func (kl *Kubelet) setNodeAddress(node *api.Node) error {
 			node.Status.Addresses = []api.NodeAddress{
 				{Type: api.NodeLegacyHostIP, Address: ipAddr.String()},
 				{Type: api.NodeInternalIP, Address: ipAddr.String()},
+				hostnameAddress,
```

Comment:
inline `api.NodeAddress{Type: api.NodeHostName, Address: kl.GetHostname()}` to make it clear what is being returned